src/api/endpoint/api/user.ts

```diff
@@ -23,13 +23,13 @@ export default function(route: Router, auth: IAuth, config: Config): void {
     const remoteName = req.remote_user.name;
 
     if (_.isNil(remoteName) === false && _.isNil(name) === false && remoteName === name) {
-      auth.authenticate(name, password, async function callbackAuthenticate(err, groups): Promise<void> {
+      auth.authenticate(name, password, async function callbackAuthenticate(err, user): Promise<void> {
         if (err) {
           logger.trace({ name, err }, 'authenticating for user @{username} failed. Error: @{err.message}');
           return next(ErrorCode.getCode(HTTP_STATUS.UNAUTHORIZED, API_ERROR.BAD_USERNAME_PASSWORD));
         }
 
-        const restoredRemoteUser: RemoteUser = createRemoteUser(name, groups);
+        const restoredRemoteUser: RemoteUser = createRemoteUser(name, user.groups || []);
```

Comment:
` || []` is not need it, but also not harmful, there is a check within the `createRemoteUser` method. 

```
const isGroupValid: boolean = Array.isArray(pluginGroups);
  const groups = (isGroupValid ? pluginGroups : []).concat([ROLES.$ALL, ROLES.$AUTH, ROLES.DEPRECATED_ALL, ROLES.DEPRECATED_AUTH, ROLES.ALL]);
```